asteroid/masknn/attention.py

```diff
@@ -122,7 +124,22 @@ def __init__(
         self.bidirectional = bidirectional
         self.dropout = dropout
 
-        self.in_norm = norms.get(norm_type)(in_chan)
+        self.mha_in_dim = ceil(self.in_chan, self.n_heads) * self.n_heads
+        if self.in_chan % self.n_heads != 0:
+            warnings.warn(
+                f"DPTransformer input dim ({self.in_chan}) is not a multiple of the number of "
+                f"heads ({self.n_heads}). Adding extra linear layers to accomodate "
+                f"(input [{self.in_chan} x {self.mha_in_dim}], "
+                f"output [{self.mha_in_dim} x {self.in_chan}])"
+            )
+            self.input_layer = nn.Linear(self.in_chan, self.mha_in_dim)
+            self.output_layer = nn.Linear(self.mha_in_dim, self.in_chan)
+        else:
+            self.input_layer = None
+            self.output_layer = None
+
+        self.in_norm = norms.get(norm_type)(self.mha_in_dim)
+        self.ola = DualPathProcessing(self.chunk_size, self.hop_size)
```

Comment:
@popcornell do you have an argument to keep the `DualPathProcessing` definition in the `forward`?

Reply:
Sorry saw this now. No my bad this was an error actually

Reply:
Thanks !